saleor/graphql/product/schema.py

```diff
@@ -453,9 +453,12 @@ def resolve_products(
     ):
         if sort_field_from_kwargs(kwargs) == ProductOrderField.RANK:
             # sort by RANK can be used only with search filter
-            if not search_string_in_kwargs(kwargs):
+            if not search_string_in_kwargs(kwargs) and not search:
```

Comment:
I guess this solution is OK, but we have two identical definitions of `search_string_in_kwargs` in Saleor; the other one is in the orders module. I suppose the bug happens there as well. 

We could fix it in all places if we had one reusable function `search_string_in_kwargs` and we changed the signature to remove `search=None` and get it from kwargs:

```
def resolve_products(_root, info: ResolveInfo, *, channel=None, **kwargs):
    search = kwargs.get("search")
```

And fixed the `search_string_in_kwargs` function:
```
def search_string_in_kwargs(kwargs: dict) -> bool:
    return bool(kwargs.get("filter", {}).get("search", "").strip()) or "search" in kwargs
```

Reply:
I added these changes to products module. Orders don't have`where` filters implemented yet,